composables/rotki-sponsorship/use-payment-tokens.ts

```diff
@@ -0,0 +1,80 @@
+import type { PaymentToken } from '~/composables/rotki-sponsorship/types';
+import { get, set } from '@vueuse/core';
+import { ETH_ADDRESS } from '~/composables/rotki-sponsorship/config';
+import { useLogger } from '~/utils/use-logger';
+
+export function usePaymentTokens() {
+  const paymentTokens = ref<PaymentToken[]>([]);
+  const isLoading = ref<boolean>(false);
+  const error = ref<string | null>(null);
+
+  const logger = useLogger('payment-tokens');
+
+  function formatPrice(price: string): string {
+    // Remove unnecessary trailing zeros after decimal point
+    const num = parseFloat(price);
+    if (isNaN(num))
+      return price;
+
+    // Convert to string and remove scientific notation if needed
+    let formatted = num.toFixed(18);
+
+    // Remove trailing zeros after decimal point
+    formatted = formatted.replace(/\.?0+$/, '');
+
+    // If it's a whole number, add .0 to maintain decimal format
+    if (!formatted.includes('.')) {
+      formatted += '.0';
+    }
+
+    return formatted;
+  }
+
+  async function fetchPaymentTokens(): Promise<void> {
+    set(isLoading, true);
+    set(error, null);
+
+    try {
+      const response = await $fetch<PaymentToken[]>('/webapi/nfts/payment-tokens/');
+
+      // Sort tokens to ensure ETH appears first
+      const sortedTokens = [...response].sort((a, b) => {
+        // ETH should always be first
+        if (a.address === ETH_ADDRESS)
+          return -1;
+        if (b.address === ETH_ADDRESS)
+          return 1;
+        // Keep the original order for other tokens
+        return 0;
+      });
+
+      set(paymentTokens, sortedTokens);
+      logger.info(`Fetched ${response.length} payment tokens`);
+    }
+    catch (error_) {
+      logger.error('Error fetching payment tokens:', error_);
+      set(error, 'Failed to fetch payment tokens');
+      throw error_;
+    }
+    finally {
+      set(isLoading, false);
+    }
+  }
+
+  const getTokenBySymbol = computed<(symbol: string) => PaymentToken | undefined>(() => (symbol: string) => get(paymentTokens).find(token => token.symbol === symbol));
+
+  const getPriceForTier = computed<(symbol: string, tier: 'bronze' | 'silver' | 'gold') => string | undefined>(() => (symbol: string, tier: 'bronze' | 'silver' | 'gold') => {
```

Comment:
Wouldn't it make more sense to extract a type alias for the tier? just to be safe?